src/components/PopUpWithForm.js

```diff
@@ -7,17 +7,29 @@ export default class PopUpWithForm extends PopUp {
     this._handleFormSubmit = handleFormSubmit;
     this._config = config;
     this._form = this._popup.querySelector(this._config.formSelector);
-    this.inputEl = this._popUpElement.querySelectorAll(".modal__input");
-    this.inputValues = document.querySelectorAll(this._config.inputSelector);
+    this.inputEl = this._popup.querySelectorAll(".modal__input");
+    this.inputValues = {};
+    this._submitButton = this._popup.querySelector(".modal__button");
+  }
+
+  setLoading(isLoading) {
+    if (isLoading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = "Save";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```